File: Fibonacci/Java/FibonacciTask.java
import java.util.concurrent.ForkJoinPool;
import java.util.concurrent.RecursiveTask;
class FibonacciTask extends RecursiveTask<Integer> {
    private final int n;

    FibonacciTask(int n) {
        this.n = n;
    }

    @Override
     protected Integer compute() {
        if (n == 0) {
            return 0;
        } else if (n == 1) {
            return 1;
        } else {
            FibonacciTask f1 = new FibonacciTask(n - 1);
            FibonacciTask f2 = new FibonacciTask(n - 2);

            f2.fork();
            return f1.compute() + f2.join();
        }
    }

    public static void main(String[] args) {
        long start_time = System.nanoTime();
        ForkJoinPool pool = new ForkJoinPool();
        System.out.println("第" + args[0] + "項" +pool.invoke(new FibonacciTask(Integer.parseInt(args[0]))));
        long end_time = System.nanoTime();
        //System.out.println(end_time-start_time);
        System.out.println("time = " + (Math.pow(10,-9)*(end_time - start_time)) + " [sec]");
    }
}

File: Fibonacci/Python/FibonacciTask.py
import time
import sys
from multiprocessing import Pool

def func(n):
    if n <2:
            return n
    else:
            return func(n-2) + func(n-1)

argv = int(sys.argv[1])
stratTime = time.time()
if __name__ == "__main__":
        p = Pool(2)
        p.map(func, range(argv))
#print('第{0}項 = {1}'.format(argv,func(argv)))
func(argv)
endTime = time.time() #プログラムの終了時刻
runTime = endTime - stratTime #処理時間
print('time = {:.10f}sec'.format(runTime)) #処理時間を表示
